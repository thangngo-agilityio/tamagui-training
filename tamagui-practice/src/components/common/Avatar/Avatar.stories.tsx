import { Meta, StoryObj } from '@storybook/react';

// Component
import Avatar from '.';

const meta: Meta<typeof Avatar> = {
  title: 'Common/Avatar',
  tags: ['autodocs'],
  component: Avatar,

  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

type Story = StoryObj<typeof Avatar>;

export const Default: Story = {
  args: {
    src: 'https://img-cdn.pixlr.com/image-generator/history/65bb506dcb310754719cf81f/ede935de-1138-4f66-8ed7-44bd16efc709/medium.webp',
    alt: 'image',
  },
};
