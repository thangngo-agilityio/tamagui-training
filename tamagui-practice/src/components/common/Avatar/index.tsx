import { Box, Image } from '@chakra-ui/react';

type TAvatarProps = {
  src?: string;
  alt?: string;
  width?: string;
  height?: string;
};

const Avatar = ({
  src = '/images/avatar-default.svg',
  alt = 'Avatar',
  width = '40px',
  height = '40px',
  ...props
}: TAvatarProps) => (
  <Box as="div" w={width} h={height} borderRadius="lg" {...props}>
    <Image
      w="100%"
      h="100%"
      borderRadius="xl"
      objectFit="cover"
      src={src}
      alt={alt}
    />
  </Box>
);

export default Avatar;
