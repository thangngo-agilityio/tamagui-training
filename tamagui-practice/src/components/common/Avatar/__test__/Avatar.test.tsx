import { render } from '@testing-library/react';

// Component
import Avatar from '..';

describe('Avatar test case', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <Avatar
        src="https://img-cdn.pixlr.com/image-generator/history/65bb506dcb310754719cf81f/ede935de-1138-4f66-8ed7-44bd16efc709/medium.webp"
        alt="image"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
