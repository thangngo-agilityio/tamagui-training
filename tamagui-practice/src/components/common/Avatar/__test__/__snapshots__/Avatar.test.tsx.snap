// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar test case Should render correctly 1`] = `
<div>
  <div
    class="css-xax5g4"
  >
    <img
      alt="image"
      class="chakra-image css-4odhbz"
      src="https://img-cdn.pixlr.com/image-generator/history/65bb506dcb310754719cf81f/ede935de-1138-4f66-8ed7-44bd16efc709/medium.webp"
    />
  </div>
</div>
`;
